configuration-files/README.md

```diff
@@ -0,0 +1,292 @@
+* [Configuration files](#configuration-files)
+    * [Split by topic](#split-by-topic)
+        * [The base](#the-base)
+        * [API](#api)
+        * [Result](#result)
+    * [Different Access-Control](#different-access-control)
+        * [Base](#base)
+        * [Dockerfile](#dockerfile)
+        * [Stage environment file](#stage-environment-file)
+        * [Production environment file](#production-environment-file)
+
+# Configuration files
+
+Couper gets configured by a [configuration-file](https://github.com/avenga/couper/tree/master/docs#configuration-file) and
+with the [latest 1.9 release](https://github.com/avenga/couper/releases/latest) Couper has an option to handle multiple
+configuration-files: the configuration-directory. This directory can be passed as an argument along with a configuration-file
+or as standalone argument.
+
+This enables you to split up a Couper configuration by content or even by environment. This is solved by [merging](https://github.com/avenga/couper/blob/master/docs/MERGE.md) the top-level
+declarations and replacing nested ones. Details are explained in our [merge documentation](https://github.com/avenga/couper/blob/master/docs/MERGE.md).
+
+## Split by topic
+
+Let's say we have some api endpoints which could be extracted out of a base configuration which just describes basics
+like files, spa serving or default environment variables.
+
+### The base
+
+We start with a simple SPA configuration and some file serving:
+
+```hcl
+server {
+  files {
+    document_root = "/htdocs"
+  }
+
+  spa {
+    bootstrap_file = "/htdocs/index.html"
+    paths = ["/", "/app"]
+  }
+
+  set_response_headers = {
+    x-service = env.SERVICE_NAME
+  }
+}
+
+defaults {
+  environment_variables = {
+    SERVICE_NAME = "example"
+  }
+}
+```
+
+### API
+
+The API part describes two api services and their related environment default values for e.g. local development:
+
+```hcl
+server {
+    api "serviceA" {
+        base_path = "/api/v1/service-a"
+        endpoint "/**" {
+            proxy {
+                url = "http://${env.SERVICE_A_ORIGIN}/"
+            }
+        }
+    }
+}
+
+defaults {
+    environment_variables = {
+        SERVICE_A_ORIGIN = "http://localhost:8080"
+        SERVICE_NAME = "service-a"
+    }
+}
+```
+
+#### Environment Variables
+
+The `environment_variables` attribute within the `defaults` block has an exception during the merge process. The map-value
+will be merged by key instead of the whole map. This allows to override or add specific environment defaults.
```

Comment:
```suggestion
The `environment_variables` map within the `defaults` block is handled differently during the merge process: Its values
are merged one by one by their key instead of replacing the whole map. This allows to override or add specific environment defaults.
```